src/setuptools_scm/__init__.py

```diff
@@ -46,22 +46,31 @@ def dump_version(
     template: str | None = None,
 ) -> None:
     assert isinstance(version, str)
-    target = os.path.normpath(os.path.join(root, write_to))
-    ext = os.path.splitext(target)[1]
+    target = Path(root) / Path(write_to)
+    ext = target.suffix
     template = template or TEMPLATES.get(ext)
     from ._trace import trace
 
     trace("dump", write_to, version)
     if template is None:
         raise ValueError(
             "bad file format: '{}' (of {}) \nonly *.txt and *.py are supported".format(
-                os.path.splitext(target)[1], target
+                ext,
+                target,
             )
         )
     version_tuple = _version_as_tuple(version)
 
-    with open(target, "w") as fp:
-        fp.write(template.format(version=version, version_tuple=version_tuple))
+    if ext == ".py":
+        with target.open("w") as fp:
+            fp.write(
+                template.format(version=version, version_tuple=version_tuple).replace(
+                    "'", '"'
```

Comment:
this pattern still can break examples

all it needs is a example with a required string thats getting broken by this

this replacement is only safe if it validates before and after generates the same ast

its far safer to have a  str wrapper that adds a format type thats returns a black safe string for tuples or strs